examples/step-57/step-57.cc

```diff
@@ -0,0 +1,879 @@
+    // interpolated data.
+    initialize_system();
+    present_solution = tmp;
+  }
+
+  // @sect4{StationaryNavierStokes::search_initial_guess}
+  //
+  // This function will provide us with an initial guess by using a
+  // continuation method as we discussed in the introduction. The Reynolds
+  // number is increased step-by-step until we reach the target value. By
+  // experiment, the solution to Stokes is good enough to be the initial guess
+  // of NSE with Reynolds number 1000 so we start there.  To make sure the
+  // solution from previous problem is close enough to the next one, the step
+  // size must be small enough.
+  template <int dim>
+  void StationaryNavierStokes<dim>::search_initial_guess(double step_size)
+  {
+    const double target_Re = 1.0/viscosity;
+
+    bool is_initial_step = true;
+
+    for (double Re=1000.0; Re < target_Re; Re = std::min(Re+step_size, target_Re))
+      {
+        viscosity = 1/Re;
+        std::cout << "*****************************************" << std::endl;
+        std::cout << " Searching for initial guess with Re = " << Re << std::endl;
+        std::cout << "*****************************************" << std::endl;
+
+        newton_iteration(1e-12, 50, 0, is_initial_step, false);
+        is_initial_step = false;
+      }
+  }
+
+  // @sect4{StationaryNavierStokes::output_results}
+  // This function is the same as in step-22.
+  template <int dim>
+  void StationaryNavierStokes<dim>::output_results (const unsigned int output_index)  const
+  {
+    std::vector<std::string> solution_names (dim, "velocity");
+    solution_names.push_back ("pressure");
+
+    std::vector<DataComponentInterpretation::DataComponentInterpretation>
+    data_component_interpretation
+    (dim, DataComponentInterpretation::component_is_part_of_vector);
+    data_component_interpretation
+    .push_back (DataComponentInterpretation::component_is_scalar);
+    DataOut<dim> data_out;
+    data_out.attach_dof_handler (dof_handler);
+    data_out.add_data_vector (present_solution, solution_names,
+                              DataOut<dim>::type_dof_data,
+                              data_component_interpretation);
+    data_out.build_patches ();
+
+    std::ostringstream filename;
+    filename << 1/viscosity
+             << "-solution-"
+             << Utilities::int_to_string (output_index, 4)
+             << ".vtk";
+
+    std::ofstream output (filename.str().c_str());
+    data_out.write_vtk (output);
+  }
+
+  // @sect4{StationaryNavierStokes::process_solution}
+  // In our test case, we do not know the analytical solution. This function
+  // outputs the velocity components along x=0.5 and y from 0 to 1 so they
+  // can be compared with data from the literature.
+  template <int dim>
+  void StationaryNavierStokes<dim>::process_solution(unsigned int refinement)
+  {
+    std::ostringstream filename;
+    filename << (1/viscosity) << "-line-" << refinement << ".txt";
```

Comment:
Write `1.0/viscosity`. Do you need to worry that the result may not be an exact integer?